src/client.ts

```diff
@@ -639,6 +640,24 @@ in an environmental variable named FAUNA_SECRET or pass it to the Client\
         JSON.stringify(response.headers),
       );
 
+      // Receiving a 200 with no body/content indicates an issue with core router
+      if (
+        response.status === 200 &&
+        response.body.length === 0 &&
+        response.headers["content-length"] === "0"
+      ) {
+        throw new ServiceInternalError(
```

Comment:
This is an error from Fauna's service; communcating that core issued a specific kind of error in the service.

ProtocolError would be more appropriate.